consensus/hotstuff/validator/validator_test.go

```diff
@@ -537,11 +641,173 @@ func (qs *QCSuite) TestQCSignatureInvalidFormat() {
 // this error as an invalid QC / invalid block, i.e. it should _not_ return an `InvalidBlockError`.
 func (qs *QCSuite) TestQCEmptySigners() {
 	*qs.verifier = mocks.Verifier{}
-	qs.verifier.On("VerifyQC", mock.Anything, qs.qc.SigData, qs.block).Return(
+	qs.verifier.On("VerifyQC", mock.Anything, qs.qc.SigData, qs.block.View, qs.block.BlockID).Return(
 		fmt.Errorf("%w", model.NewInsufficientSignaturesErrorf("")))
 
 	// the Validator should _not_ interpret this as a invalid QC, but as an internal error
-	err := qs.validator.ValidateQC(qs.qc, qs.block)
+	err := qs.validator.ValidateQC(qs.qc)
 	assert.True(qs.T(), model.IsInsufficientSignaturesError(err)) // unexpected error should be wrapped and propagated upwards
 	assert.False(qs.T(), model.IsInvalidBlockError(err), err, "should _not_ interpret this as a invalid QC, but as an internal error")
 }
+
+func TestValidateTC(t *testing.T) {
+	suite.Run(t, new(TCSuite))
+}
+
+type TCSuite struct {
+	suite.Suite
+	participants flow.IdentityList
+	signers      flow.IdentityList
+	indices      []byte
+	block        *model.Block
+	tc           *flow.TimeoutCertificate
+	committee    *mocks.DynamicCommittee
+	verifier     *mocks.Verifier
+	validator    *Validator
+}
+
+func (s *TCSuite) SetupTest() {
+
+	// create a list of 10 nodes with 1-weight each
+	s.participants = unittest.IdentityListFixture(10,
+		unittest.WithRole(flow.RoleConsensus),
+		unittest.WithWeight(1),
+	)
+
+	// signers are a qualified majority at 7
+	s.signers = s.participants[:7]
+
+	var err error
+	s.indices, err = signature.EncodeSignersToIndices(s.participants.NodeIDs(), s.signers.NodeIDs())
+	require.NoError(s.T(), err)
+
+	rand.Seed(time.Now().UnixNano())
+	view := uint64(int(rand.Uint32()) + len(s.participants))
```

Comment:
```suggestion
	view := uint64(int(rand.Uint32()))
```
Any reason to add `len(participants)` to view? It's not a problem really, just a bit confusing.